include/sourcepp/detail/StringUtils.h

```diff
@@ -22,4 +25,195 @@ void trim(std::string& s, std::string_view c);
+        {
+            if(this->selfOwning)
+                return this->owningString;
+            return std::string(string);
+        }
+    }
+
+    bool operator==( const std::string& str ) const
+    {
+        return string == str;
+    }
+
+    bool operator==( const std::string_view& str ) const
+    {
+        return string == str;
+    }
+
+    bool operator==( const char* str) const
+    {
+        return string == str;
+    }
+
+    UtilStringView& operator=(const char* str)
+    {
+        owningString = str;
+        string = owningString;
+        selfOwning = true;
+        return *this;
+    }
+
+    UtilStringView& operator=(std::string str)
+    {
+        owningString = std::move(str);
+        string = owningString;
+        selfOwning = true;
+        return *this;
+    }
+
+    UtilStringView& operator=(std::string_view str)
+    {
+        string = str;
+        return *this;
+    }
+
+    UtilStringView& operator<<(const char* str)
+    {
+        this->append(str);
+        return *this;
+    }
+
+    UtilStringView& operator<<(std::string_view str)
+    {
+        this->append(str);
+        return *this;
+    }
+
+    UtilStringView& operator<<(const std::string& str)
+    {
+        this->append(str);
+        return *this;
+    }
+
+    static UtilStringView make_self_owning(std::string_view str)
+    {
+        UtilStringView selfOwningView = {};
+        selfOwningView.owningString = str;
+        selfOwningView.string = selfOwningView.owningString;
+        selfOwningView.selfOwning = true;
+        return selfOwningView;
+    }
+
+    static UtilStringView make_view(std::string_view str)
+    {
+        UtilStringView string_view = {};
+        string_view.string = str;
+        return string_view;
+    }
+
+    static UtilStringView make_view(const std::string& str)
+    {
+        UtilStringView string_view = {};
+        string_view.string = str;
+        return string_view;
+    }
+
+    static UtilStringView make_view(const char* str)
+    {
+        UtilStringView string_view = {};
+        string_view.string = str;
+        return string_view;
+    }
+
+    [[nodiscard]] size_t length() const
+    {
+        return this->string.length();
+    }
+
+    [[nodiscard]] const char* data() const
+    {
+        return this->string.data();
+    }
+
+    [[nodiscard]] std::string get_string() const
+    {
+        return std::string(string);
+    }
+
+    [[nodiscard]] bool ends_with(std::string_view str) const
+    {
+        return this->string.ends_with(str);
+    }
+
+    std::string_view& append(const std::string_view & str)
+    {
+        if(!this->selfOwning) {
+            this->owningString = string;
+            this->selfOwning = true;
+        }
+        this->owningString.append(str);
+        this->string = std::string_view(owningString);
+        return this->string;
+    }
+
+    [[nodiscard]] bool isSelfOwning() const
+    {
+        return selfOwning;
+    }
+
+};
```

Comment:
This entire class can be replaced with a char pointer and a bool, and if implemented correctly it's exactly as safe as this class